metricbeat/module/docker/cpu/helper.go

```diff
@@ -80,27 +80,30 @@ func getNewCpu(stats *dc.Stats) CPURaw {
 
 func (c *CPUService) PerCpuUsage(stats *dc.Stats) common.MapStr {
 	var output common.MapStr
-	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOLdCpu(stats).PerCpuUsage) {
+	if cap(getNewCpu(stats).PerCpuUsage) == cap(getOldCpu(stats).PerCpuUsage) {
```

Comment:
Why `cap()` and not `len()`? https://github.com/elastic/beats/pull/2709#discussion_r82501857

The range loop that follows produces iteration values from 0 up to `len(x)-1`. So even though their capacities are equal that does not guarantee that their `len()`'s are equal. This could lead to an `index out of range` runtime error.